apps/meteor/app/oembed/server/providers.ts

```diff
@@ -11,13 +13,20 @@ class Providers {
 		this.providers = [];
 	}
 
-	static getConsumerUrl(provider: OEmbedProvider, url: string): string {
+	static getConsumerUrl(provider: OEmbedProvider, url: string): string | undefined {
+		if (!provider.endPoint) {
```

Comment:
Is there any possibility of endPoint being undefined here? If so, we should update that to reflect on the typings.

Reply:
Yes, 
I have modified the provider definition.
- made endpoint optional
- introduced optional headerOverrides
This will help us in cases when we want to modify the headers without adding endpoint.
Hence the server will call the default parsed url with custom headers. 
Some urls returns meta tags depending on user agents.

PS. I have modified the types already.